jscomp/test/mt.ml

```diff
@@ -61,14 +61,25 @@ let is_mocha () =
   | _ -> false
 (* assert -- raises an AssertionError which mocha handls better
 *)
-let from_suites name (suite :  (string * ('a -> unit)) list) =
-  match Array.to_list Node.Process.process##argv with
-  | cmd :: _ ->
-    if is_mocha () then
-      describe name (fun [@bs] () ->
-          List.iter (fun (name, code) -> it name code) suite)
 
-  | _ -> ()
+let from_suites =
+[%bs.raw {|
+function from_suites(name, suite) {
+  var match = $$Array.to_list(Process.argv);
+  if (match && is_mocha(undefined)) {
+    describe(name, (function () {
+            return List.iter((function (param) {
+                          var partial_arg = param[1];
+                          it(param[0], (function () {
```

Comment:
Somehow these tests depend on begin zero arity: `function () { ... }`. Using raw to not change the generated .js.